### ISSUE TITLE
A few questions/suggestions about the code

### ISSUE THREAD

**Issue by alexekkis** [rykov8/ssd_keras#59]
First of all, thanks a lot for porting this to Keras. I'd have a few questions about the code and would greatly appreciate your feedback:

1. Why are you performing the L2 normalization only once, before the predictions off of `conv4_3`? Shouldn't the normalization be performed before each prediction layer?
2. On the other hand one could ask: Why are you performing the L2 normalization at all? ParseNet's motivation for introducing such normalization was that they needed to equalize features of different scales in order to use them in subsequent layers. The situation is different in SSD, you're not concatenating features of varying scales - all convolutional classifier layers are independent of each other and each individual conv classifier will learn suitable weights to manage whatever features it receives as input.
3. Is the PriorBox layer necessary? I don't see you using the prior box information in your `compute_loss` function, so I was wondering why it needs to be a layer in the graph. I see that `BBoxUtility` uses the prior box information in order to assign prior boxes to ground truth boxes, but that happens in the batch generator outside the network, and also you seem to be loading the prior boxes from a pickle file, so it doesn't seem to be necessary as a layer as part of the graph? Correct me if the following statement is wrong: The only part of the process where the default boxes are relevant is to determine which classifier cells are responsible for which ground truth boxes (by assigning prior boxes to ground truth boxes and setting the category truth labels accordingly during training). But that's a data preprocessing matter, not a part of the computational graph. If that's not correct then I misunderstood the paper, in which case I'd be grateful to be corrected.
4. It'd be awesome if you were so kind to tell me what the last dimension of `y_true` contains, especially from `y_true[:,:,-8]` onwards. You're using `y_true[:,:,-8]` to compute `num_pos`, but I don't quite understand what data it contains.

It'd be great to hear your thoughts and thanks a lot for any feedback!

**Comment by jogundas** [comment]
1. and 2. It was implemented that way originally, too, see this [issue](https://github.com/weiliu89/caffe/issues/241) on the original SSD repository.
3. That is an implementation choice. Note that the PriorBox layers do not have any trainable parameters, as can be seen from `model.summary()`. See also this [issue](https://github.com/rykov8/ssd_keras/issues/71).
